pkg/network/usm/shared_libraries.go

```diff
@@ -105,15 +104,12 @@ type soWatcher struct {
 	registry       *soRegistry
 }
 
-type pathIdentifierSet = map[pathIdentifier]struct{}
-
 type soRegistry struct {
-	m     sync.RWMutex
-	byID  map[pathIdentifier]*soRegistration
-	byPID map[uint32]pathIdentifierSet
+	byID  sync.Map // map[pathIdentifier]*soRegistration
```

Comment:
Do we know what's the performance overhead for using the `sync.Map` type instead of a regular map with locks?

Reply:
Hard to tell, especially as there are multiple changes here